Change tidb_opt_projection_push_down default value to ON (#18346)
pingcap/docs: system-variables.md

### tidb_opt_projection_push_down <span class="version-mark">New in v6.1.0</span>

> - Scope: SESSION | GLOBAL

Reviewer: ```suggestion
- Scope: SESSION | GLOBAL
- Persists to cluster: Yes/No
```

- Applies to hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value): Yes
- Type: Boolean

> - Default value: `ON`
> - Specifies whether to allow the optimizer to push `Projection` down to the TiKV coprocessor. Currently only three types of `Projection` operators are allowed to be pushed down:
>     - The top-level expressions of the operator are all JSON query class or JSON value attribute class functions, such as `JSON_EXTRACT`, `JSON_DEPTH`.
>     - The top-level expressions of the operator are partly JSON query class or JSON value attribute class functions, and partly direct column reads.
>     - The top-level expressions of the operator are all direct column reads, and the number of columns output is less than the number of columns input.
> - Whether the `Projection` operator is finally pushed down or not depends on the optimizer's comprehensive evaluation of performance.

Reviewer: ```suggestion
- Default value: `ON`. Before v8.3.0, the default value is `OFF`.
- Specifies whether to allow the optimizer to push `Projection` down to the TiKV coprocessor. When enabled, the optimizer might push the following three types of `Projection` `Projection` operators down to TiKV:
    - The top-level expressions of the operator are all [JSON query functions](/functions-and-operators/json-functions/json-functions-search.md) or [JSON value attribute functions](/functions-and-operators/json-functions/json-functions-return.md). For example: `SELECT JSON_EXTRACT(data, '$.name') FROM users;`.
    - The top-level expressions of the operator include a mix of JSON query functions or JSON value attribute functions, and direct column reads. For example: `SELECT JSON_DEPTH(data), name FROM users;`.
    - The top-level expressions of the operator are all direct column reads, and the number of output columns is less than the number of input columns. For example: `SELECT name FROM users;`.
- The final decision to push down a `Projection` operator depends on the optimizer's comprehensive evaluation of performance.
```

### tidb_opt_projection_push_down <span class="version-mark">New in v6.1.0</span>

- Scope: SESSION | GLOBAL
- Persists to cluster: Yes
- Applies to hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value): Yes
- Type: Boolean
- Default value: `ON`. Before v8.3.0, the default value is `OFF`.

> - Specifies whether to allow the optimizer to push `Projection` down to the TiKV coprocessor. When enabled, the optimizer might push the following three types of `Projection` `Projection` operators down to TiKV:

Reviewer 2: ```suggestion
- Specifies whether to allow the optimizer to push the `Projection` operator down to the TiKV coprocessor. When enabled, the optimizer might push the following three types of `Projection` operators down to TiKV:
```

    - The top-level expressions of the operator are all [JSON query functions](/functions-and-operators/json-functions/json-functions-search.md) or [JSON value attribute functions](/functions-and-operators/json-functions/json-functions-return.md). For example: `SELECT JSON_EXTRACT(data, '$.name') FROM users;`.
    - The top-level expressions of the operator include a mix of JSON query functions or JSON value attribute functions, and direct column reads. For example: `SELECT JSON_DEPTH(data), name FROM users;`.
    - The top-level expressions of the operator are all direct column reads, and the number of output columns is less than the number of input columns. For example: `SELECT name FROM users;`.

> - The final decision to push down a `Projection` operator depends on the optimizer's comprehensive evaluation of query cost.

Reviewer 2: ```suggestion
- The final decision to push down a `Projection` operator also depends on the optimizer's comprehensive evaluation of query cost.
```

> - The final decision to push down a `Projection` operator depends on the optimizer's comprehensive evaluation of performance.

Reviewer: ```suggestion
- The final decision to push down a `Projection` operator depends on the optimizer's comprehensive evaluation of query cost.
```

> - This variable is disabled by default for TiDB clusters that are upgraded from a version earlier than v8.3.0 to v8.3.0 or later.

Reviewer: ```suggestion
- For TiDB clusters that are upgraded from a version earlier than v8.3.0 to v8.3.0 or later, this variable remains the setting before the upgrade.
```

> - For TiDB clusters that are upgraded from a version earlier than v8.3.0 to v8.3.0 or later, this variable will be set to `OFF`.

Reviewer: ```suggestion
- For TiDB clusters that are upgraded from a version earlier than v8.3.0 to v8.3.0 or later, the default value of this variable is `OFF`.
```

> - For TiDB clusters that are upgraded from a version earlier than v8.3.0 to v8.3.0 or later, this variable remains the setting before the upgrade.

Author: ```suggestion
- For TiDB clusters that are upgraded from a version earlier than v8.3.0 to v8.3.0 or later, this variable will be set to `OFF`.
```
